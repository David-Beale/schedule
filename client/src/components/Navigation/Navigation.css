.navigation-container {
  position: fixed;
  overflow: hidden;
  z-index: 10;
  top: 0;
  width: 100vw;
  height: 5vh;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  background-color: rgb(255, 255, 255);
  box-shadow: 1px 1px 8px 1px rgba(189, 189, 189, 0.2);
}
.navigation-name {
  font-family: 'Montserrat', sans-serif;
  color: #f05537;
  font-size: 1.7rem;;
  font-weight: 600;
  padding-left: 2rem;
}
.navigation-links-container {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  margin: 0 3rem;
}

.button {
  color: #111111;
  font-size: 1.2rem;
  margin: 0.2vh;
  padding: 0.3rem;
  text-align: center;
  text-decoration: none;
  cursor: pointer;
  transition: 0.5s;
}
.icon {
  padding-top: 0.2rem;
  max-width: 1.5rem;
  max-height: 1.5rem;
  padding-left: 1rem;
}
