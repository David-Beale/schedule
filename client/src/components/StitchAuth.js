import React, {
  useEffect,
  createContext,
  useContext,
  useState,
  useMemo
} from 'react';

import {
  hasLoggedInUser,
  loginAnonymous,
  loginGoogle,
  logoutCurrentUser,
  getCurrentUser,
  addAuthenticationListener,
  removeAuthenticationListener,
  handleOAuthRedirects
} from './../stitch/authentication';

const StitchAuthContext = createContext();

export function useStitchAuth() {
  const context = useContext(StitchAuthContext);
  if (!context) {
    throw new Error(`useStitchAuth must be used within a StitchAuthProvider`);
  }
  return context;
}

export function StitchAuthProvider(props) {
  const [authState, setAuthState] = useState({
    isLoggedIn: hasLoggedInUser(),
    currentUser: getCurrentUser()
  });

  useEffect(() => {
    const authListener = {
      onUserLoggedIn: (auth, loggedInUser) => {
        if (loggedInUser) {
          setAuthState((authState) => ({
            ...authState,
            isLoggedIn: true,
            currentUser: loggedInUser
          }));
        }
      },
      // eslint-disable-next-line no-unused-vars
      onUserLoggedOut: (auth, loggedOutUser) => {
        setAuthState((authState) => ({
          ...authState,
          isLoggedIn: false,
          currentUser: null
        }));
      }
    };
    addAuthenticationListener(authListener);
    handleOAuthRedirects();
    setAuthState((state) => ({ ...state }));
    return () => {
      removeAuthenticationListener(authListener);
    };
  }, []);

  // Authentication Actions
  const handleLogin = async (provider) => {
    if (!authState.isLoggedIn) {
      switch (provider) {
        case 'anonymous':
          return loginAnonymous();
        case 'google':
          console.log('hoohle!');
          return loginGoogle();
        // eslint-disable-next-line no-empty
        default: {
        }
      }
    }
  };
  const handleLogout = async () => {
    const { isLoggedIn } = authState;
    if (isLoggedIn) {
      await logoutCurrentUser();
      setAuthState({
        ...authState,
        isLoggedIn: false,
        currentUser: null
      });
    } else {
      console.log(`can't handleLogout when no user is logged in`);
    }
  };

  // We useMemo to improve performance by eliminating some re-renders
  const authInfo = useMemo(() => {
    const { isLoggedIn, currentUser } = authState;
    const value = {
      isLoggedIn,
      currentUser,
      actions: { handleLogin, handleLogout }
    };
    return value;
  }, [authState.isLoggedIn]);
  return (
    <StitchAuthContext.Provider value={authInfo}>
      {props.children}
    </StitchAuthContext.Provider>
  );
}
