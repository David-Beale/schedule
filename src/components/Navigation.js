import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/navigation.css'

function Navigation (props) {
  return (
    <div className="navigation-container">
      <Link
        to="/"
        className="button"
        >

        <div className="navigation-name">Logo</div>
      </Link>
      <div className="navigation-links-container">
        <Link
          className="button"
          to="/"
        >
          <div>Home</div>
        </Link>
        <Link
          className="button"
          to="/projects"
        >
          <div>Nav1</div>
        </Link>
        <Link
          className="button"
          to="/about"
        >
          <div>Nav2</div>
        </Link>
        <Link
          className="button"
          to="/contact"
        >
          <div>Nav3</div>
        </Link>
      </div>
    </div>
  )
}

export default Navigation