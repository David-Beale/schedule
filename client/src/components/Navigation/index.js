import Navigation from './Navigation';
import './Navigation.css';

export { Navigation as default };
