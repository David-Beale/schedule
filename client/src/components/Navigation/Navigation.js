import React from 'react';
import { Link } from 'react-router-dom';
import { useStitchAuth } from '../StitchAuth/StitchAuth';
import './Navigation.css';

function Navigation(props) {
  const { isLoggedIn, actions } = useStitchAuth();
  const toggleMenu = () => {
    const x = document.getElementById('myTopnav');
    if (x.className === "navigation-container") {
      x.className += " responsive";
    } else {
      x.className = "navigation-container";
    }
  }

  return (
    <div id='myTopnav' className="navigation-container">
      <Link to="/" id='logo' className="button">
        <div className="navigation-name">Logo</div>
      </Link>
      <div className="navigation-links-container">
        <Link className="button menu" onClick={toggleMenu} to="/">
          <div>Home</div>
        </Link>
        <Link className="button menu" onClick={toggleMenu} to="/schedule">
          <div>Schedule</div>
        </Link>
        {isLoggedIn && (
          <Link className="button menu" onClick={toggleMenu} to="/events">
            <div>Your Events</div>
          </Link>
        )}
        {isLoggedIn && (
          <Link className="button menu" onClick={toggleMenu} to="/form">
            <div>Create Event</div>
          </Link>
        )}
        <Link className="button menu" onClick={toggleMenu} to="/about">
          <div>About</div>
        </Link>
        {isLoggedIn ? (
          <Link className="button menu" onClick={toggleMenu} to="/logout">
            <div onClick={actions.handleLogout}>Logout</div>
          </Link>
        ) : (
            <Link className="button menu" onClick={toggleMenu} to="/login">
              <div>Login</div>
            </Link>
          )}
        <a className="icon" onClick={toggleMenu}>&#9776;</a>
      </div>
    </div>
  );
}

export { Navigation as default };
