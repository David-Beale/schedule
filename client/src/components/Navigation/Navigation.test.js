import Navigation from './Navigation';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import React from 'react';
import { StitchAuthProvider } from '../StitchAuth/StitchAuth';
import { BrowserRouter as Router } from 'react-router-dom';
describe('Navigation', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <StitchAuthProvider>
          <Navigation />
        </StitchAuthProvider>
      </Router>,
      div
    );
  });
  it('should render properly', () => {
    const component = render(
      <Router>
        <StitchAuthProvider>
          <Navigation />
        </StitchAuthProvider>
      </Router>
    );
    expect(component).toMatchSnapshot();
  });
});
