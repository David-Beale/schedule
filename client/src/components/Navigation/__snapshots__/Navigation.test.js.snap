// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navigation-container"
        id="myTopnav"
      >
        <a
          class="button"
          href="/"
          id="logo"
        >
          <div
            class="navigation-name"
          >
            Logo
          </div>
        </a>
        <div
          class="navigation-links-container"
        >
          <a
            class="button menu"
            href="/"
          >
            <div>
              Home
            </div>
          </a>
          <a
            class="button menu"
            href="/schedule"
          >
            <div>
              Schedule
            </div>
          </a>
          <a
            class="button menu"
            href="/about"
          >
            <div>
              About
            </div>
          </a>
          <a
            class="button menu"
            href="/login"
          >
            <div>
              Login
            </div>
          </a>
          <a
            class="icon"
          >
            ☰
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navigation-container"
      id="myTopnav"
    >
      <a
        class="button"
        href="/"
        id="logo"
      >
        <div
          class="navigation-name"
        >
          Logo
        </div>
      </a>
      <div
        class="navigation-links-container"
      >
        <a
          class="button menu"
          href="/"
        >
          <div>
            Home
          </div>
        </a>
        <a
          class="button menu"
          href="/schedule"
        >
          <div>
            Schedule
          </div>
        </a>
        <a
          class="button menu"
          href="/about"
        >
          <div>
            About
          </div>
        </a>
        <a
          class="button menu"
          href="/login"
        >
          <div>
            Login
          </div>
        </a>
        <a
          class="icon"
        >
          ☰
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
